### PR TITLE
#1291: export tests logs to PR comment

### LINKED ISSUE #1291: Build warnings/errors forwarded don't include test status
**What Needs to be Done?**

It currently forwards whether the build succeeded or not, which does not include test status. Explore expanding the comment mechanism to include ctest status from the docker container.

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1291

### CODE REVIEW

File: tests/unit/collection/test_lb.extended.cc [hunk 1]
```diff
@@ -193,7 +193,7 @@ TEST_P(TestNodeStatsDumper, test_node_stats_dumping_with_interval) {
   auto const file_path =
     fmt::format("{}/{}", theConfig()->vt_lb_stats_dir, file_name);
   auto const readPhases = getPhasesFromStatsFile(file_path.c_str());
-  EXPECT_EQ(readPhases.size(), num_phases);
+  EXPECT_EQ(readPhases.size(), num_phases+1);
```
Comment (jstrzebonski) [inline tests/unit/collection/test_lb.extended.cc:196]: That's temporary to check if tests' logs are exported correctly.

**Comment by jstrzebonski** [comment]
There are some errors after moving the whole functionality to the script. I'll fix it tomorrow.

**Comment by jstrzebonski** [comment]
OK, I moved everything related to comments creation to a bash script to simplify `yml` a little.
If you're OK with how the comments look now, I'll revert the test bug and remove debug information from azure pipeline.

@lifflander @PhilMiller @nmm0 @nlslatt @JacobDomagala @cz4rs

**Review by lifflander — APPROVED** [review]
Looks great

**Comment by JacobDomagala** [comment]
Would it be possible to include direct link to the build aswell? That'd be great to have, instead of having to click through 4 pages to see the logs. 

Assuming the Azure provides the variables for BUILD_ID, JOB_ID and TASK_ID it should be fairly easy to generate the URL as follows:
https://dev.azure.com/DARMA-tasking/DARMA/_build/results?buildId={BUILD_ID}&view=logs&j={JOB_ID}&t={TASK_ID}

And then the comment would include that link like so:
> **PR tests (gcc-7, ubuntu, mpich, trace runtime, LB)**
> Build for 14d832ded591db8bf31e6b77ea575c111b974d5e
> 
> ```
> Compilation - successful
> 
> The following tests FAILED:
>    79 - vt:*/TestNodeStatsDumper.test_node_stats_dumping_with_interval/*_proc_2 (Failed)
> ```
> 
> For details go to -> [Build Logs](https://dev.azure.com/DARMA-tasking/DARMA/_build/results?buildId=9427&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=28db5144-7e5d-5c90-2820-8676d630d9d2)

**Review by JacobDomagala — APPROVED** [review]

**Comment by jstrzebonski** [comment]
@JacobDomagala 
Yeah, that sounds reasonable.

I'll merge this one, so everyone can enjoy those super-duper comments already, and let's open another issue for that link.

**Review by cz4rs — APPROVED** [review]

**Pull request merged**